test/processStates_test.js

```diff
@@ -0,0 +1,250 @@
+goog.require('helper');
+
+describe('The processStates function', function() {
+  /**
+   * Asserts that calling processStates_ with the given arguments will result
+   * in the expected model state and expected calls to the listener.
+   *
+   * @param {Array.<Object>} states The states argument for the call to
+   *     processStates_.
+   * @param {Object} expectedModel The expected model state after the call.
+   * @param {Array.<Array.<*>>} expectedListenerCalls The expected calls made
+   *     to the helper's listener. Should be an array of arrays where each sub-
+   *     array represents the arguments passed to the listener during a call.
+   */
+  function assertProcessStates(states, expectedModel, expectedListenerCalls) {
+    let MockHelper = function() {
+      this.model_ = {};
+      this.unprocessed_ = [];
+      this.executingListener_ = false;
+      this.listenerCalls_ = [];
+      this.listener_ = function() {
+        this.listenerCalls_.push(
+            jQuery.extend(true, [], [].slice.call(arguments, 0)));
+      };
+      this.abstractModelInterface_ = helper.buildAbstractModelInterface_(this);
+    };
+
+    function doAssert(skipListener) {
+      MockHelper.prototype = new DataLayerHelper([]);
+      let helper = new MockHelper();
+      DataLayerHelper.prototype.processStates_.call(helper, states,
+          skipListener);
+      expect(helper.model_).toEqual(expectedModel);
+      expect(helper.listenerCalls_).
+          toEqual(skipListener ? [] : expectedListenerCalls);
+      expect(helper.unprocessed_).toEqual([]);
+      expect(helper.executingListener_).toEqual(false);
```

Comment:
In type_test.js we are using toBe() instead of toEqual() which works since toBe() handles primitives just fine.

I suggest we stay consistent and either change type_test.js to use toEqual() since it handles both objects and primitives or change processStates_test.js to use toEqual() with objects and toBe() with primitives.

Reply:
Good catch, I will stay with the latter choice.